### ISSUE TITLE
Public repo with no sources available?

### ISSUE THREAD

**Issue by xucian** [Rem0o/FanControl.Releases#1143]
"Unknown publisher" notice is really a downer for such popular software. Is there a public code I can compile myself?
I'm assuming not, but the code should be public for us to trust it's malware-free

**Comment by antt0n** [comment]
Without even the question of malware, I think it could support the development of the project by making PR.

**Comment by Rem0o** [comment]
I didn't buy a certificate from a certificate authority, since this is a hobby project, not a commercial product, hence the "unknown publisher". 

As for open-source, the core dlls that actually do stuff to your components are open-source, the UI is not.
You can read a discussion on the subject here: https://github.com/Rem0o/FanControl.Releases/issues/89

TLDR: The app/ui is not open-source. If you don't trust it, don't run it.

**Comment by tfgstudios** [comment]
@Rem0o 
I fail to see how someone stealing this code would benefit from it (especially since we're discussing UI-only code) while being detrimental to the original author.

"If you don't trust it, don't run it.". That's not something someone that has nothing to hide would say. If I'd made a software that I'm proud of or think it's useful for others, I would've encouraged people to use it and also try to provide some means of assuring the code is legit (even in case of closed-source), instead of saying "take it or leave it".
Though I don't accuse the author, I just put things in perspective.

IMO, software should have at least one of the following:
- signed
- open-sourced

Note that I'm not contesting anyone's right to choose their own source code's visibility/license -- that's everyone's right and I respect it.

I respect your work. Let's open-source it

**Comment by inoperable** [comment]
@Rem0o you shouldn't try to convince anyone about the closed vs open - this is merit at least and upping a bin blob here
is just adding extra load to yourself. You should really consider putting the whole repository here so people can poke at it - some people far smarter then you and me might be among them, one can always learn from those people. 

That's my not so humble opinion. You do what you thinks is best.

(btw. im a bit tired of shuffling fan control sources through ilspy)

I sponsor you (although that's a merit too)

**Comment by inoperable** [comment]
Seems our benefactor can't handle unconformable questions or has a peculiar inbox filter that haven't reached yet a critical mass.

**Comment by Rem0o** [comment]
@inoperable  My stand on the subject hasn't changed.  I get this question a lot since I use Github as a distribution/issue platform, but the answer is the same. 

The software is donationware, closed sources, open for plugins, based on open-sources libraries.

**Comment by inoperable** [comment]
You forgot about obfuscation that's brin recently added to it

**Comment by yokunjon** [comment]
@Rem0o Then don't use github as your distrubition and issues platform. The norm here is sources being available, open-source or licensed, doesn't matter, majority of people share source code here. I'd call locating issues and distrubition of your project here without source code is shady practice. It isn't written anywhere except this issue that this project is closed source, you should've stated it with big, bold sentence, at start of the README. There are lots of people coming here just for the software, which you are aware of, as you shared the video mentioned your project; most of them are not software engineers or have no knowledge about how github works. Due to this being on github, they might think that this is open-source. Even though you appended "Releases" to the repo name, it could be understood as there is another repo with the source code. It could be understood as this is release only mirror from another service.

Yes, it is still their responsibility take care of their downloads; but you should also act ethical and provide people the info they should know in the first place. I don't really see any difference between hiding it compared to not sharing it; results are same.

**Comment by Rem0o** [comment]
@yokunjon  I get your point.  I'm not trying to deceive anybody here on purpose. I see Github as a tool, and that tool has served this project quite well.  It allows me to reference directly stuff in other libraries in the issues section, monitor releases, promote open-source plugins and so on. 

I've realized that the readme being the "front-page" for FanControl might be a bit misleading, hence why recently I've been working on a website: https://getfancontrol.com/ 

I may tone the README down to the bare minimum so that this becomes only a "utility" page with download + issues, not much promotion of the software itself. 

Now I can understand that with Github, there can be expectations of available sources, and that expectation here can bring deception. I can see that's the case for some people, and I'm sorry about that. 

I think that explicitly mentioning the closed-source nature of the software is a fair ask, so I will integrate that near the top of the README.

**Comment by yokunjon** [comment]
Yeah, I don't think your intent is malicious. Thanks, really appreciated!

**Comment by xucian** [comment]
I still wonder why won't you share the sources. Forking? I, as a user+developer won't use a new 400-star repo instead of this one, even if it's more up-to-date. I think this fear is unfounded, and here's why:
I expect the original author to keep maintaining the repo, so I'll keep using it. If they seen to have abandoned the repo, only then other forks start taking traction.
Example: [SD webui](https://github.com/AUTOMATIC1111/stable-diffusion-webui) - 5.6 forks, I'd never use any other one than the original fork. If I have a very specific need, I'd use a fork only for a single project where I have that need.
There's a 'good enough' signing certificate [here](https://comodosslstore.com/codesigning.aspx?gclid=Cj0KCQiAic6eBhCoARIsANlox85141RRBsW0xsWCuwjPdE2nTRoGVucVmWUlOMOmyT7Cjqbw5TKFv9saAkn6EALw_wcB), for example. Surely you could make 85$/yr with a PayPal donate button.

But yes, no one can say you're forced to open source it. But ethics say, and I think most of other users agree, that if sources are closed, you need a trusted 3rd party to be responsible for the software's safety, i.e. have a publisher license for that platform (see my [above conditions](https://github.com/Rem0o/FanControl.Releases/issues/1143#issuecomment-1266707224)).

**Comment by quinndiggity** [comment]
Fixed in https://github.com/Rem0o/FanControl.Releases/pull/1629
